postgresql/Containerfile.ubi

```diff
@@ -0,0 +1,37 @@
+FROM registry.access.redhat.com/ubi9@sha256:ed84f34cd929ea6b0c247b6daef54dd79602804a32480a052951021caf429494
+
+# PostgreSQL image for OpenShift.
+# Volumes:
+#  * /var/lib/pgsql/data   - Database cluster for PostgreSQL.
+# Environment:
+#  * $POSTGRESQL_USER     - Database user name.
+#  * $POSTGRESQL_PASSWORD - User's password.
+#  * $POSTGRESQL_DATABASE - Name of the database to create.
+#  * $POSTGRESQL_ADMIN_PASSWORD (Optional) - Password for the 'postgres' administrator user.
+
+ENV POSTGRESQL_VERSION=16 \
+    POSTGRESQL_PREV_VERSION=15 \
+    HOME=/var/lib/pgsql
+
+EXPOSE 5432
+
+ENV PATH="/usr/pgsql-${POSTGRESQL_VERSION}/bin:${PATH}"
+
+RUN  { yum -y module enable "postgresql:${POSTGRESQL_VERSION}" || :; } && \
+    dnf install -y https://download.postgresql.org/pub/repos/yum/reporpms/EL-9-x86_64/pgdg-redhat-repo-latest.noarch.rpm && \
```

Comment:
This looks like it'll get in the way of making this a hermetic build.

Reply:
From Konflux POV, postgresql has to be a separate component regardless. We should be building on top of an RH build of PostgreSQL, likely https://catalog.redhat.com/software/containers/rhel9/postgresql-13/61a60a80bfd4a5234d596288?architecture=amd64&image=66612c572bc32cd731fea67b, instead of going directly to postgresql.org.

Reply:
We can't build it on top of  `rhel9/postgresql-13` because of the comments Ben mentioned above

Reply:
We can't be installing packages off the internet either. 
If we can't use rhel9/postgresql-13 directly, we should be building a 'sister' image to it.  There are PostgreSQL packages built in the RH downstream (I'm looking at the constituent packages of  rhel9/postgresql-13):
```
postgresql-13.14-1.el9_3.x86_64	PostgreSQL client programs
postgresql-contrib-13.14-1.el9_3.x86_64	Extension modules distributed with PostgreSQL
postgresql-private-libs-13.14-1.el9_3.x86_64	The shared libraries required only for this build of PostgreSQL server
postgresql-server-13.14-1.el9_3.x86_64	The programs needed to create and run a PostgreSQL server
```

Reply:
but we can install the same postgres packages it installs